x/oracle/abci.go

```diff
@@ -56,6 +63,16 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) error {
 			if err = k.SetExchangeRateWithEvent(ctx, ballotDenom.Denom, exchangeRate); err != nil {
 				return err
 			}
+
+			// Stamp rate every stamp period
+			if isPeriodLastBlock(ctx, params.StampPeriod) {
+				k.AddHistoricPrice(ctx, ballotDenom.Denom, exchangeRate)
```

Comment:
Where stamp price will be used? Why do we need it? Is the use-case already drafted and validated?

Reply:
Stamping is just what we're calling adding the historic price to the store since that price is stamped by what block it was calculated in